### DISCUSSION TITLE
errors trying to build llvm from sources using ninja

### DISCUSSION THREAD

**Discussion by PavelKotov1** [intel/llvm#4740] [Q&A]
Hello, dear friends!
I am trying to build from sources and install llvm on windows 10. Installed all the prerequisites first.
Then: git clone https://github.com/llvm/llvm-project.git
cd llvm-project
mkdir build
cd build
cmake –G Ninja -DLLVM_ENABLE_PROJECTS=clang ../llvm
ninja
The output of the last command is as follows:

`[1/4267] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regstrlcpy.c.obj
[2/4267] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regfree.c.obj
[3/4267] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regexec.c.obj
[4/4267] Building C object lib/Support/CMakeFiles/LLVMSupport.dir/regerror.c.obj
[5/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/xxhash.cpp.obj
[6/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Z3Solver.cpp.obj
[7/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/TrigramIndex.cpp.obj
[8/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/ToolOutputFile.cpp.obj
[9/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/TimeProfiler.cpp.obj
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/TimeProfiler.cpp.obj 
C:\MinGW\bin\c++.exe  -DGTEST_HAS_RTTI=0 -D_DEBUG -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Ilib/Support -IC:/llvm-project/llvm/lib/Support -Iinclude -IC:/llvm-project/llvm/include -isystem "C:/Program Files (x86)/GnuWin32/include" -Wa,-mbig-obj -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -Wmisleading-indentation -g   -std=c++14  -fno-exceptions -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/TimeProfiler.cpp.obj -MF lib\Support\CMakeFiles\LLVMSupport.dir\TimeProfiler.cpp.obj.d -o lib/Support/CMakeFiles/LLVMSupport.dir/TimeProfiler.cpp.obj -c C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:31:13: error: 'mutex' in namespace 'std' does not name a type
 static std::mutex Mu;
             ^~~~~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp: In member function 'void llvm::TimeTraceProfiler::write(llvm::raw_pwrite_stream&)':
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:128:21: error: 'mutex' is not a member of 'std'
     std::lock_guard<std::mutex> Lock(Mu);
                     ^~~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:128:21: error: 'mutex' is not a member of 'std'
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:128:31: error: template argument 1 is invalid
     std::lock_guard<std::mutex> Lock(Mu);
                               ^
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:128:38: error: 'Mu' was not declared in this scope
     std::lock_guard<std::mutex> Lock(Mu);
                                      ^~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:128:33: warning: unused variable 'Lock' [-Wunused-variable]
     std::lock_guard<std::mutex> Lock(Mu);
                                 ^~~~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp: In function 'void llvm::timeTraceProfilerCleanup()':
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:275:19: error: 'mutex' is not a member of 'std'
   std::lock_guard<std::mutex> Lock(Mu);
                   ^~~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:275:19: error: 'mutex' is not a member of 'std'
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:275:29: error: template argument 1 is invalid
   std::lock_guard<std::mutex> Lock(Mu);
                             ^
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:275:36: error: 'Mu' was not declared in this scope
   std::lock_guard<std::mutex> Lock(Mu);
                                    ^~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:275:31: warning: unused variable 'Lock' [-Wunused-variable]
   std::lock_guard<std::mutex> Lock(Mu);
                               ^~~~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp: In function 'void llvm::timeTraceProfilerFinishThread()':
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:284:19: error: 'mutex' is not a member of 'std'
   std::lock_guard<std::mutex> Lock(Mu);
                   ^~~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:284:19: error: 'mutex' is not a member of 'std'
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:284:29: error: template argument 1 is invalid
   std::lock_guard<std::mutex> Lock(Mu);
                             ^
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:284:36: error: 'Mu' was not declared in this scope
   std::lock_guard<std::mutex> Lock(Mu);
                                    ^~
C:/llvm-project/llvm/lib/Support/TimeProfiler.cpp:284:31: warning: unused variable 'Lock' [-Wunused-variable]
   std::lock_guard<std::mutex> Lock(Mu);
                               ^~~~
[10/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Timer.cpp.obj
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/Timer.cpp.obj 
C:\MinGW\bin\c++.exe  -DGTEST_HAS_RTTI=0 -D_DEBUG -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Ilib/Support -IC:/llvm-project/llvm/lib/Support -Iinclude -IC:/llvm-project/llvm/include -isystem "C:/Program Files (x86)/GnuWin32/include" -Wa,-mbig-obj -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -Wmisleading-indentation -g   -std=c++14  -fno-exceptions -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/Timer.cpp.obj -MF lib\Support\CMakeFiles\LLVMSupport.dir\Timer.cpp.obj.d -o lib/Support/CMakeFiles/LLVMSupport.dir/Timer.cpp.obj -c C:/llvm-project/llvm/lib/Support/Timer.cpp
In file included from C:/llvm-project/llvm/lib/Support/Timer.cpp:24:0:
C:/llvm-project/llvm/include/llvm/Support/Mutex.h:29:12: error: 'recursive_mutex' in namespace 'std' does not name a type
       std::recursive_mutex impl;
            ^~~~~~~~~~~~~~~
C:/llvm-project/llvm/include/llvm/Support/Mutex.h: In member function 'bool llvm::sys::SmartMutex<mt_only>::lock()':
C:/llvm-project/llvm/include/llvm/Support/Mutex.h:35:11: error: 'impl' was not declared in this scope
           impl.lock();
           ^~~~
C:/llvm-project/llvm/include/llvm/Support/Mutex.h: In member function 'bool llvm::sys::SmartMutex<mt_only>::unlock()':
C:/llvm-project/llvm/include/llvm/Support/Mutex.h:48:11: error: 'impl' was not declared in this scope
           impl.unlock();
           ^~~~
C:/llvm-project/llvm/include/llvm/Support/Mutex.h: In member function 'bool llvm::sys::SmartMutex<mt_only>::try_lock()':
C:/llvm-project/llvm/include/llvm/Support/Mutex.h:62:18: error: 'impl' was not declared in this scope
           return impl.try_lock();
                  ^~~~
[11/4267] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Triple.cpp.obj
**ninja: build stopped: subcommand failed.**
`
Could somebody help me figure out what is going on? I am absolutely new to the subject so please be understanding.
Thanks in advance!

**Comment by Michoumichmich** [comment]
Hi. I think that you're in the wrong place to ask for support as you're referring to `https://github.com/llvm/llvm-project.git`. However, I'd just recommend you to check the dependencies needed to build LLVM. You could try using `clang-11` or something newer. If that doesn't solve the issue, you should consider opening a ticket on the llvm project. 

What about: https://github.com/randombit/botan/issues/2039 ?

**Comment by alexbatashev** [comment]
Duplicate of https://github.com/intel/llvm/issues/4734
Bugs for upstream LLVM project must be submitted directly to LLVM: https://llvm.org/docs/HowToSubmitABug.html
